### PR TITLE
Using different active and owner keys

### PR DESCRIPTION
**Pull request by dafuga**

### CODE REVIEW

File: packages/snap/src/lib/keyDeriver.ts [hunk 1]
```diff
@@ -22,36 +22,82 @@ async function getKeyDeriver(chain: ChainDefinition) {
...
+ *
+ * @param keyIndex - The index of the address to derive (must not be 0).
+ * @returns The active private key.
+ * @throws If the key tree is not initialized or if index 0 is used.
+ */
+export async function deriveActivePrivateKey(
+  chain: ChainDefinition,
+  keyIndex = 1,
+): Promise<PrivateKey> {
+  if (keyIndex === 0) {
```
Comment (aaroncox) [inline packages/snap/src/lib/keyDeriver.ts:98]: I don't think this is a strong enough check.

If I pass in `keyIndex = undefined` to this function, the value won't be equal to 0 - so it'll bypass this and then pass `undefined` to `derivePrivateKey`, which will default to 0 and return the owner key.

Maybe we need to write it like this:

```ts
if (Number(keyIndex) > 0) {
  return derivePrivateKey(chain, keyIndex);
}
throw new Error('Invalid key index');
```

We just need to make sure there's no way to access index 0 from this call.

**Review by aaroncox — APPROVED** [review]

File: packages/snap/src/lib/keyDeriver.ts (hunk 1 above)
Comment (dafuga) [reply packages/snap/src/lib/keyDeriver.ts:98]: if you pass `undefined` then the default `1` will be used 😄

Comment (dafuga) [reply packages/snap/src/lib/keyDeriver.ts:98]: if you want, I'll change the default of `derivePrivateKey` to 1 so the owner key isn't returned by default

Comment (dafuga) [reply packages/snap/src/lib/keyDeriver.ts:98]: Done 👍

**Pull request merged**